chain-config/src/validator_rules.rs

```diff
@@ -17,18 +18,33 @@ pub struct TokenIdAmountPair<M: ManagedTypeApi> {
 #[multiversx_sc::module]
 pub trait ValidatorRulesModule: setup_phase::SetupPhaseModule + events::EventsModule {
     fn is_new_config_valid(&self, config: &SovereignConfig<Self::Api>) -> Option<&str> {
+        if let Some(additional_stake) = config.opt_additional_stake_required.clone() {
+            for stake in additional_stake {
+                require!(stake.token_id.is_valid_esdt_identifier(), INVALID_TOKEN_ID);
+            }
+        }
+
         if config.min_validators <= config.max_validators {
             None
         } else {
             Some(INVALID_MIN_MAX_VALIDATOR_NUMBERS)
         }
     }
 
+    #[payable]
     #[endpoint(register)]
     fn register(&self, new_validator: ValidatorInfo<Self::Api>) {
```

Comment:
I believe the validations should be:
- validate if enabled/disabled
- validate additional stake 
- require_validator_not_registered
- the rest is fine

Reply:
I think you can make another PR for `unregister` where you check enable/disabled and return the additional stake

Reply:
Also the endpoint for enabling/disabling registrations which is done with 2 steps - register + update(or change_status or something similar)

Reply:
Also the endpoint to read the genesis validators